### PR TITLE
Add simdjson/0.2.1

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **simdjson/0.2.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Fixes https://github.com/conan-io/wishlist/issues/221

### CODE REVIEW

**Comment by Croydon** [comment]
Please add a `Fixes https://github.com/conan-io/wishlist/issues/221` into the pull request description, so that the issues gets closed when this gets merged. Thanks!

File: recipes/simdjson/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class SimdjsonConan(ConanFile):
+    name = "simdjson"
+    description = "Parsing gigabytes of JSON per second"
+    topics = ("conan", "json", "parser", "simd", "format")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "Ahttps://github.com/lemire/simdjson"
```
Comment (SSE4) [inline recipes/simdjson/all/conanfile.py]:
```suggestion
    homepage = "https://github.com/lemire/simdjson"
```

**Comment by SSE4** [comment]
seems like CMake crashes? it terminates via **1073741819** which is access violation error:
```
error MSB3073: "C:\Program Files\CMake\bin\cmake.exe" -E __create_def ...
...
exited with code -1073741819.
```

**Comment by theirix** [comment]
@SSE4 yes, seems cmake is crashing. I am trying to debug this error on a Windows VM (VS 2019, x64, latest cmake from conan_cmake_installer) and it is reproduced locally but I have no idea why. It is related to post-build action (cmake property `WINDOWS_EXPORT_ALL_SYMBOLS 1`) when a def file is produced based on object files. No proper CMake tickets.

The simdjson is rapidly developed, I'll try master and report a bug upstream.

Maybe we just ban shared builds for simdjson on Windows?

**Comment by SSE4** [comment]
@theirix is it latest CMake release available?
if so, let's report it to the CMake, and disable the shared build leaving the comment to the newly created ticket.

**Comment by theirix** [comment]
The problem was non-obvious. It was a conflict of enabled LTO and exporting symbols. CMake command `__export_def` cannot extract data from object files generated by MSVC with LTO information as well as `dumpbin.exe` (I think nobody can). I have disabled it for shared MSVC builds.

File: recipes/simdjson/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        cmake = self._configure_cmake()
+        cmake.install()
+        # remove unneeded directories
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+
+        self.copy("license", src=self._source_subfolder, dst="licenses", ignore_case=True, keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.libs = ['simdjson']
```
Comment (uilianries) [inline recipes/simdjson/all/conanfile.py]:
```suggestion
        self.cpp_info.libs = ['simdjson']
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["m"]
```

ldd lib/libsimdjson.so.0
	linux-vdso.so.1 (0x00007ffc82f45000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f80dd6a6000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f80dd560000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f80dd546000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f80dd37f000)
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007f80dd8e8000)

File: recipes/simdjson/all/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.9)
```
[the remark this answers is not in the document]
Comment (theirix) [reply recipes/simdjson/all/CMakeLists.txt]: Unfortunately, upstream project requires CMake 3.9 (LTO checks) so our wrapper should too. Is that correct?

**Comment by theirix** [comment]
I created a ticket https://github.com/lemire/simdjson/issues/451 in the upstream project to track our workaround in the future.

File: recipes/simdjson/all/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/simdjson/all/CMakeLists.txt]: Okay, I didn't read your previous comment, thanks for reporting.

**Pull request merged**